modules/tooling/src/main/java/org/mule/runtime/module/tooling/api/connectivity/ConnectionResult.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.tooling.api.connectivity;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+import static org.mule.runtime.module.tooling.api.connectivity.ConnectionResult.Status.FAILURE;
+import static org.mule.runtime.module.tooling.api.connectivity.ConnectionResult.Status.SUCCESS;
+
+import java.util.Optional;
+
+/**
+ * Result of doing connectivity testing in a mule component
+ *
+ * @since 4.0
+ */
+public class ConnectionResult
```

Comment:
wouldn't it be better to use org.mule.runtime.api.connection.ConnectionValidationResult? These two classes seem very alike